main.tf

```diff
@@ -38,13 +39,22 @@ resource "aws_default_security_group" "quortex" {
 }
 
 resource "aws_vpc_ipv4_cidr_block_association" "secondary" {
-  for_each   = var.vpc_secondary_cidrs
+  for_each   = toset([for index, az in var.availability_zones : cidrsubnet(var.vpc_secondary_cidr, 2, index)])
   vpc_id     = aws_vpc.quortex.id
   cidr_block = each.value
 }
 
 resource "aws_subnet" "quortex" {
-  for_each = var.subnets
+  for_each = merge([
+    for key, subnet in var.subnets : {
+      for index, az in var.availability_zones : "${key}-${data.aws_region.current.name}${az}" => {
+        "availability_zone" = "${data.aws_region.current.name}${az}",
+        "cidr"              = cidrsubnet(subnet.cidr, 2, index),
```

Comment:
You could want newbits it to be 1 if there is 2 AZ... or even 3 if there is more than 4 AZ.
Anyway, it should not be hardcoded.